lib/chef/knife/bootstrap_windows_base.rb

```diff
@@ -426,29 +416,6 @@ def warn_chef_config_secret_key
         WARNING
         ui.info "* " * 40
       end
-
-      # We allow the user to specify the desired architecture of Chef to install or we default
-      # to whatever the target system is.  We assume that we are only bootstrapping 1 node at a time
-      # so we don't need to worry about multipe responses from this command.
-      def set_target_architecture(bootstrap_architecture)
-        session_results = relay_winrm_command("echo %PROCESSOR_ARCHITECTURE%")
-        if session_results.empty? || session_results[0].stdout.strip.empty?
-          raise "Response to 'echo %PROCESSOR_ARCHITECTURE%' command was invalid: #{session_results}"
-        end
-        current_architecture = session_results[0].stdout.strip == "X86" ? :i386 : :x86_64
-
-        if bootstrap_architecture.nil?
-          architecture = current_architecture
-        elsif bootstrap_architecture == :x86_64 && current_architecture == :i386
-          raise "You specified bootstrap_architecture as x86_64 but the target machine is i386. A 64 bit program cannot run on a 32 bit machine."
-        else
-          architecture = bootstrap_architecture
-        end
-
-        # The windows install script wants i686, not i386
-        architecture = :i686 if architecture == :i386
```

Comment:
I think we've standardized on i386 in omnitruck, but there's some back-compat support for i686 too. it looks like #msi_url in knife-windows hits the omnitrucks API rather than constructing a URL (which is correct) so it shouldn't matter. 

https://github.com/chef/omnitruck/blob/master/app.rb#L314

But maybe there's some old functionality I'm missing. we should test a few versions either way, like something before 12.9. Before 12.9 IIRC we always server 32-bit packages out of stable, but you could get 64bit packages out of current if they existed. For 12.9+ you should get whatever you ask for.